src/NUnitTestAdapter/NUnitTestFilterBuilder.cs

```diff
@@ -1,44 +1,52 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 using Microsoft.VisualStudio.TestPlatform.ObjectModel;
 using NUnit.Engine;
 
 namespace NUnit.VisualStudio.TestAdapter
 {
     public class NUnitTestFilterBuilder
     {
-        private ITestFilterService _filterService;
+        private readonly ITestFilterService _filterService;
 
         public static readonly TestFilter NoTestsFound = new TestFilter("<notestsfound/>");
+        private readonly IAdapterSettings settings;
+        private readonly ITestLogger logger;
 
-        public NUnitTestFilterBuilder(ITestFilterService filterService)
+        public NUnitTestFilterBuilder(ITestFilterService filterService,IAdapterSettings settings, ITestLogger log)
         {
-            if (filterService == null)
-                throw new NUnitEngineException("TestFilterService is not available. Engine in use is incorrect version.");
-
-            _filterService = filterService;
+            logger = log;
+            this.settings = settings;
+            _filterService = filterService ?? throw new NUnitEngineException("TestFilterService is not available. Engine in use is incorrect version.");
         }
 
         public TestFilter ConvertTfsFilterToNUnitFilter(TfsTestFilter tfsFilter, List<TestCase> loadedTestCases)
         {
             var filteredTestCases = tfsFilter.CheckFilter(loadedTestCases);
             var testCases = filteredTestCases as TestCase[] ?? filteredTestCases.ToArray();
             //TestLog.Info(string.Format("TFS Filter detected: LoadedTestCases {0}, Filterered Test Cases {1}", loadedTestCases.Count, testCases.Count()));
-            return MakeTestFilter(testCases);
+            return MakeTestFilter(testCases,tfsFilter);
         }
 
-        public TestFilter MakeTestFilter(IEnumerable<TestCase> testCases)
+        public TestFilter MakeTestFilter(IEnumerable<TestCase> testCases, TfsTestFilter tfsFilter)
         {
-            if (testCases.Count() == 0)
+            if (!testCases.Any())
                 return NoTestsFound;
 
-            ITestFilterBuilder filterBuilder = _filterService.GetTestFilterBuilder();
-
-            foreach (TestCase testCase in testCases)
-                filterBuilder.AddTest(testCase.FullyQualifiedName);
-
+            var filterBuilder = _filterService.GetTestFilterBuilder();
+            if (tfsFilter!=null && settings.UseTestCaseFilterConverter)
+            {
+                var nunitfilter = new VsTest2NUnitFilterConverter(tfsFilter.TfsTestCaseFilterExpression.TestCaseFilterValue);
+                filterBuilder.SelectWhere(nunitfilter.ToString());
```

Comment:
Since there is no error detection, It seems like there is an underlying assumption here that the converter will always produce a valid where clause. Do we know that's true for all possible values of the original filter?